### PR TITLE
feat: move docs out of extras into their own group

### LINKED ISSUE #165: `mkdocs` dependencies should be declared as a dependency group, not extras
Per [past discussion](https://fulcrumgenomics.slack.com/archives/C04BQ9RMJ2G/p1718200843607199?thread_ts=1718200462.992179&cid=C04BQ9RMJ2G), I think it would make more sense to declare `mkdocs` and friends as a dependency group, rather than as extras. 

Extras are made available via pip's bracketed syntax (e.g. `pip install fgpyo[pysam]`) and imo should be limited to optional dependencies _for the end-user_.

Dependency groups are only available via `poetry install` (e.g. `poetry install --with docs`) and imo should be used for optional or required developer dependencies. (Required dependencies should be declared in the `dev` group, and optionals in other groups, as appropriate.)

For `mkdocs`, these dependencies could be declared in a `docs` group. However, if our `ci/check.sh` script requires `mkdocs build` to pass, I would suggest including these in the `dev` group.

### PR DESCRIPTION
**Pull request by nh13**
Fixes #165

### CODE REVIEW

**Comment by nh13** [comment]
Using a separate group for better granularity, versus lumping it all into the dev group

File: pyproject.toml [hunk 1]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
...
+mkdocs-autorefs = { version = ">=0.5.0" }
+mkdocs-include-markdown-plugin = { version = ">=6.0.1" }
+mkdocs-material = { version = ">=9.2.8" }
+mkdocs-table-reader-plugin = { version = ">=2.0.1" }
+mkdocs = { version = ">=1.5.2" }
+mkdocs-gen-files = { version = ">=0.5.0" }
+mkdocs-literate-nav = { version = ">=0.6.1" }
+mkdocs-section-index = { version = ">=0.3.9" }
+mkdocstrings-python = { version = ">=1.6.2" }
+mkdocstrings = { version = ">=0.23.0" }
```
Comment (msto) [inline pyproject.toml:40]: I think these can be removed from the package dependencies now, if they're provided via the `docs` group

```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
...
+black = {version = ">=24.4.2"}
```
Comment (msto) [inline pyproject.toml:41]:
```suggestion
```

Similarly I think `black` can be removed as a package dependency and made a `dev` dependency.

File: pyproject.toml [hunk 3]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
...
 
 [tool.poetry.dev-dependencies]
```
Comment (msto) [inline pyproject.toml:65]:
```suggestion
[tool.poetry.group.dev.dependencies]
```

`dev-dependencies` is [deprecated](https://python-poetry.org/docs/managing-dependencies/)

> Poetry will slowly transition away from the dev-dependencies notation which will soon be deprecated, so it’s advised to migrate your existing development dependencies to the new group notation.

File: pyproject.toml [hunk 4]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
 pysam = ">=0.22.1"
 pytest = ">=7.4.0"
```
Comment (msto) [inline pyproject.toml:29]:
```suggestion
```

I think `pytest` should also be a dev dependency, not a package dependency

File: pyproject.toml [hunk 5]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
...
+mkdocs-include-markdown-plugin = { version = ">=6.0.1" }
+mkdocs-material = { version = ">=9.2.8" }
+mkdocs-table-reader-plugin = { version = ">=2.0.1" }
+mkdocs = { version = ">=1.5.2" }
+mkdocs-gen-files = { version = ">=0.5.0" }
+mkdocs-literate-nav = { version = ">=0.6.1" }
+mkdocs-section-index = { version = ">=0.3.9" }
+mkdocstrings-python = { version = ">=1.6.2" }
+mkdocstrings = { version = ">=0.23.0" }
+black = {version = ">=24.4.2"}
```
Comment (msto) [inline pyproject.toml:63]:
```suggestion
```

I think this should be included in `dev` instead of `docs`

File: pyproject.toml [hunk 6]
```diff
@@ -28,38 +28,39 @@ attrs = ">=19.3.0"
...
 
 [tool.poetry.dev-dependencies]
 ruff = "0.4.8"
```
Comment (msto) [inline pyproject.toml:66]: Could you please add `black` to the dev dependencies?

**Review by msto — CHANGES_REQUESTED** [review]
@nh13 If the `docs` dependencies are not required dev dependencies, would you be open to removing `mkdocs build` from the `ci/check.sh` script? I think that script should be runnable with only dev dependencies.

File: pyproject.toml (hunk 5 above)
Comment (nh13) [reply pyproject.toml:63]: `mkdocstrings` and friends need `black` for some reason!

**Review by msto — APPROVED** [review]
lgtm!

**Review by tfenne — APPROVED** [review]
LGTM

**Pull request merged**